packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -206,6 +212,22 @@ function getNonChildrenInnerMarkup(props) {
   return null;
 }
 
+function flattenTopLevelChildren(children: mixed): FlatReactChildren {
+  if (!React.isValidElement(children)) {
+    return toArray(children);
+  }
+  const element = ((children: any): ReactElement);
+  if (element.type !== REACT_FRAGMENT_TYPE) {
```

Comment:
This is just an optimization right? Not semantic meaning. I wonder if this is just slowing down the normal case where this is `true` since the check will happen again then.

Reply:
Is there a way to do less checks? I think that every way we re-write this, there still ends up being 3 checks. https://github.com/facebook/react/pull/10783/files/48bcaa7198ef5d97bfc78badd1636a836a076cda..27312d0ebbc3ebbdaa8dc48e6f70119847267389#r147533483